File: src/frontend/src/api/notification.ts
/**
 * 站内通知 API
 */

import request from '@/utils/http'
import type { HttpClient } from '@/utils/http'

const httpClient = request as HttpClient

const PREFIX = '/api/v1/notifications'

export interface NotificationItem {
  id: string
  user_id: number
  type: 'alert' | 'ai_done' | 'system' | 'test'
  title: string
  content: string | null
  link: string | null
  is_read: boolean
  created_at: string
}

export interface NotificationListResp {
  total: number
  items: NotificationItem[]
  page: number
  page_size: number
}

/** 通知列表（可筛选 is_read） */
export const fetchNotifications = (params?: { page?: number; page_size?: number; is_read?: boolean }) => {
  return httpClient.get<Http.BaseResponse<NotificationListResp>>({
    url: `${PREFIX}`,
    params: { page: 1, page_size: 20, ...params },
    keepFullResponse: true
  })
}

/** 未读数 */
export const fetchUnreadCount = () => {
  return httpClient.get<Http.BaseResponse<{ count: number }>>({
    url: `${PREFIX}/unread-count`,
    keepFullResponse: true
  })
}

/** 标记单条已读 */
export const markNotificationRead = (id: string) => {
  return httpClient.post<Http.BaseResponse<NotificationItem>>({
    url: `${PREFIX}/${id}/read`,
    keepFullResponse: true
  })
}

/** 全标已读 */
export const markAllNotificationsRead = () => {
  return httpClient.post<Http.BaseResponse<{ updated: number }>>({
    url: `${PREFIX}/mark-all-read`,
    keepFullResponse: true
  })
}

File: src/backend/app/schemas/notification.py
"""Notification Schemas

站内通知相关的请求/响应模型。
"""

from datetime import datetime
from typing import List, Optional
from uuid import UUID

from pydantic import BaseModel, Field


# ============== Request ==============


class NotificationTestRequest(BaseModel):
    """管理员手动测试通知"""

    title: str = Field(default="测试通知", max_length=200)
    content: str = Field(default="这是一条测试通知", max_length=1000)
    link: Optional[str] = Field(default=None, max_length=500)


# ============== Response ==============


class NotificationOut(BaseModel):
    """单条通知出参"""

    id: UUID
    user_id: int
    type: str
    title: str
    content: Optional[str]
    link: Optional[str]
    is_read: bool
    created_at: datetime

    class Config:
        from_attributes = True


class NotificationListResponse(BaseModel):
    """通知列表分页响应"""

    total: int
    items: List[NotificationOut]
    page: int
    page_size: int


class UnreadCountOut(BaseModel):
    """未读数"""

    count: int


class MarkAllReadOut(BaseModel):
    """全标已读影响行数"""

    updated: int
